test/validation/policy/PolicyCRUDFuzz.t.sol

```diff
@@ -255,59 +236,200 @@ abstract contract PolicyCRUDFuzzTest is RulesEngineCommon {
-            assertEq(callingFunctions_.length, selectors.length, "selector length mismatch");
-            assertEq(callingFunctionIds_.length, functionIds.length, "function id length mismatch");
-            assertEq(ruleIds_.length, _ruleIds.length, "rule id length mismatch");
-            for (uint i; i < _ruleIds.length; i++) {
-                assertEq(ruleIds_[i].length, functionAmount, "rule id length mismatch");
+        if (!shouldRevert) {
+            (bytes4[] memory callingFunctions_, uint256[][] memory ruleIds_) = RulesEnginePolicyFacet(address(red)).getPolicy(policyId);
+            assertEq(callingFunctions_.length, functionAmount, "selector length mismatch");
+            assertEq(ruleIds_.length, functionAmount, "rule id length mismatch");
+            for (uint i; i < ruleIds_.length; i++) {
+                console2.log("i", i);
+                console2.log("ruleIds_[0][0]", ruleIds_[0][0]);
+                assertEq(ruleIds_.length, functionAmount, "rule id length mismatch");
+                assertEq(ruleIds_[i].length, 1, "rule id length mismatch");
                 RuleStorageSet memory ruleStorage = RulesEngineRuleFacet(address(red)).getRule(policyId, ruleIds_[i][0]);
                 assertEq(ruleStorage.rule.instructionSet.length, 7, "instruction set length mismatch");
             }
         }
     }
+
+    function testPolicy_updatePolicy_identicalElementsInCalldataArrayBytes4(uint seedSig, uint multiplier) public {
+        uint sigAmount = 4;
+        TestFacetUtils facet = new TestFacetUtils();
+        bytes4[] memory sigs = new bytes4[](sigAmount);
+        for (uint i; i < sigAmount; i++) {
+            unchecked {
+                sigs[i] = bytes4(bytes32(seedSig * multiplier * (i + 1)));
+            }
+        }
+        /// the following condition is basically a representation of what the algorithm does
+        if (
+            (sigs[0] == sigs[1] || sigs[0] == sigs[2] || sigs[0] == sigs[3]) ||
+            (sigs[1] == sigs[2] || sigs[1] == sigs[3]) ||
+            sigs[2] == sigs[3]
+        ) vm.expectRevert("Duplicates not allowed");
+        facet.checkDuplicatesBytes4(sigs);
+    }
+
+    function testPolicy_updatePolicy_identicalElementsInCalldataArrayBytes32(uint seedSig, uint multiplier) public {
+        uint sigAmount = 4;
+        TestFacetUtils facet = new TestFacetUtils();
+        bytes32[] memory sigs = new bytes32[](sigAmount);
+        for (uint i; i < sigAmount; i++) {
+            unchecked {
+                sigs[i] = bytes32(seedSig * multiplier * (i + 1));
+            }
+        }
+        /// the following condition is basically a representation of what the algorithm does
+        if (
+            (sigs[0] == sigs[1] || sigs[0] == sigs[2] || sigs[0] == sigs[3]) ||
+            (sigs[1] == sigs[2] || sigs[1] == sigs[3]) ||
+            sigs[2] == sigs[3]
+        ) vm.expectRevert("Duplicates not allowed");
+        facet.checkDuplicatesBytes32(sigs);
```

Comment:
Think I see now, just wanted clarification the reason we're doing the byte shifting is because duplicate calling functions are no longer allowed.